crates/bevy_pbr/src/material.rs

```diff
@@ -147,6 +147,14 @@ pub trait Material: AsBindGroup + Send + Sync + Clone + TypeUuid + Sized + 'stat
     ) -> Result<(), SpecializedMeshPipelineError> {
         Ok(())
     }
+
+    /// Returns a list of vertex attributes required by this `Material`.
+    ///
+    /// The default implementation of this method returns an empty `Vec`.
+    #[inline]
+    fn required_vertex_attributes() -> Vec<MeshVertexAttribute> {
```

Comment:
I was going to say that I feel like API-wise this should return `&[MeshVertexAttribute]` so that it is possible to avoid repeated Vec allocations in cases where meshes/materials are modified every frame.